pmg/sync_soundcloud.py

```diff
@@ -0,0 +1,61 @@
+import logging
+from pmg import db
+from pmg.models import *
+from pmg.models.soundcloud import SoundcloudTrack
+from sqlalchemy import desc
+import soundcloud
+
+logger = logging.getLogger(__name__)
+
+
+def get_unstarted_query():
+    return db.session.query(File, Event.title.label('event_title')) \
+                     .join(EventFile) \
+                     .join(Event) \
+                     .outerjoin(SoundcloudTrack) \
+                     .filter(SoundcloudTrack.file_id == None) \
+                     .filter(File.file_mime.like('audio/%')) \
+                     .order_by(desc(Event.date))
+
+
+def get_unstarted_count(q):
+    return q.count()
+
+
+def get_unstarted_batch(q):
+    return q.limit(app.config['MAX_SOUNDCLOUD_BATCH']).all()
+
+
+def sync():
+    q = get_unstarted_query()
+    logging.info("Audio files yet to be uploaded to SoundCloud: %d" % get_unstarted_count(q))
+    batch = get_unstarted_batch(q)
+    logging.info("Uploading %d files to SoundCloud" % get_unstarted_count(q))
+    for file, event_title in batch:
+        upload_file(file, event_title)
+
+
+def upload_file(file, event_title):
```

Comment:
I'd put most of the logic in this file onto the SoundcloudTrack class. It's directly related, @classmethods are useful ways of scoping logic into a module. eg. ``SoundcloudTrack.new_from_event_file(event_file)`` or something.